rsocket-core/src/main/java/io/rsocket/core/DefaultClientRSocketFactory.java

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright 2002-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.rsocket.core;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufAllocator;
+import io.rsocket.AbstractRSocket;
+import io.rsocket.ConnectionSetupPayload;
+import io.rsocket.DuplexConnection;
+import io.rsocket.Payload;
+import io.rsocket.RSocket;
+import io.rsocket.RSocketFactory;
+import io.rsocket.SocketAcceptor;
+import io.rsocket.frame.ResumeFrameFlyweight;
+import io.rsocket.frame.SetupFrameFlyweight;
+import io.rsocket.frame.decoder.PayloadDecoder;
+import io.rsocket.internal.ClientServerInputMultiplexer;
+import io.rsocket.internal.ClientSetup;
+import io.rsocket.keepalive.KeepAliveHandler;
+import io.rsocket.lease.LeaseStats;
+import io.rsocket.lease.Leases;
+import io.rsocket.lease.RequesterLeaseHandler;
+import io.rsocket.lease.ResponderLeaseHandler;
+import io.rsocket.plugins.DuplexConnectionInterceptor;
+import io.rsocket.plugins.PluginRegistry;
+import io.rsocket.plugins.Plugins;
+import io.rsocket.plugins.RSocketInterceptor;
+import io.rsocket.plugins.SocketAcceptorInterceptor;
+import io.rsocket.resume.ExponentialBackoffResumeStrategy;
+import io.rsocket.resume.InMemoryResumableFramesStore;
+import io.rsocket.resume.ResumableFramesStore;
+import io.rsocket.resume.ResumeStrategy;
+import io.rsocket.transport.ClientTransport;
+import io.rsocket.util.EmptyPayload;
+import io.rsocket.util.MultiSubscriberRSocket;
+import java.time.Duration;
+import java.util.Objects;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+import reactor.core.Disposable;
+import reactor.core.publisher.Mono;
+import reactor.util.retry.Retry;
+
+/**
+ * Default implementation of {@link RSocketFactory.ClientRSocketFactory} that can be instantiated
+ * directly or through the shortcut {@link RSocketFactory#connect()}.
+ */
+public class DefaultClientRSocketFactory implements RSocketFactory.ClientRSocketFactory {
```

Comment:
Generally, all the comments for the API might be applied

Reply:
I'll do that in a separate PR for #769.